src/algebra/dual_number.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import algebra.triv_sq_zero_ext
+
+/-!
+# Dual numbers
+
+The dual numbers over `R` are of the form `a + bε`, where `a` and `b` are typically elements of a
+commutative ring `R`, and `ε` is a symbol satisfying `ε^2 = 0`. They are a special case of
+`triv_sq_zero_ext R M` with `M = R`.
+
+## Notation
+
+* `𝔻[R]` as a shorthand for `dual_number R`, in the `dual_number` locale.
```

Comment:
Would it work to make `R[ε]` the notation? That's very common notation in algebraic geometry. I guess Lean shouldn't have a problem with the suffix notation, right?

Reply:
That notation is probably ok; I only picked this notation because it looks like ` ℍ[R]`

Reply:
I've gone ahead and changed to that notation.